#! /bin/sh -e

# DP: Make it so that a missing /etc/ld.so.conf does not cause an error
# DP: message, unless --verbose is enabled. This keeps the debian
# DP: installer from barfing during bootstrap of the system.
# DP:
# DP: Updated by gotom, the previous one seems being wrong because it supresses
# DP: _all_ warnings about "can't open configuration file". otoh, I introduce
# DP: newer patches.  it should be checked using chroot and should be contacted
# DP: to the upstream.
# DP: Date: (Updated 2005-01-02 gotom)

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

2004-10-25  GOTO Masanori  <gotom@debian.or.jp>

        * elf/ldconfig.c: Don't print error when default ld.so.conf
          is not existed.

Index: elf/ldconfig.c
===================================================================
RCS file: /cvs/glibc/libc/elf/ldconfig.c,v
retrieving revision 1.47
diff -u -r1.47 ldconfig.c
--- elf/ldconfig.c	10 Aug 2004 04:04:32 -0000	1.47
+++ elf/ldconfig.c	2 Jan 2005 08:46:26 -0000
@@ -972,7 +972,8 @@
 
   if (file == NULL)
     {
-      error (0, errno, _("Can't open configuration file %s"), canon);
+      if (strcmp(canon, LD_SO_CONF) != 0 || opt_verbose)
+	error (0, errno, _("Can't open configuration file %s"), canon);
       if (canon != filename)
 	free ((char *) canon);
       return;

# previous patch.
#--- elf/ldconfig.c	2003-07-08 23:26:27.000000000 +0900
#+++ elf/ldconfig.c.debian	2003-07-08 23:29:43.000000000 +0900
#@@ -920,26 +920,24 @@
# {
#   FILE *file = NULL;
#   char *line = NULL;
#-  const char *canon;
#+  const char *canon = filename;
#   size_t len = 0;
#+  int file_fd;
# 
#   if (opt_chroot)
#     {
#       canon = chroot_canon (opt_chroot, filename);
#-      if (canon)
#-	file = fopen (canon, "r");
#-      else
#+      if (!canon)
# 	canon = filename;
#     }
#-  else
#-    {
#-      canon = filename;
#-      file = fopen (filename, "r");
#-    }
#+
#+  if ((file_fd = open(canon, O_RDONLY | O_EXCL, 0022)) != -1)
#+    file = fdopen (file_fd, "r");
# 
#   if (file == NULL)
#     {
#-      error (0, errno, _("Can't open configuration file %s"), canon);
#+      if (opt_verbose)
#+	error (0, errno, _("Can't open configuration file %s"), canon);
#       if (canon != filename)
# 	free ((char *) canon);
#       return;
